hd-docs/system-architecture-overview.md

```
@@ -0,0 +1,278 @@
+Frontend Display
+```
+
+### Real-Time Communication
+```
+Frontend ←→ Server-Sent Events ←→ Main Server ←→ WebSocket ←→ 
+LLM Provider (streaming)
+```
+
+## Technology Stack Deep Dive
+
+### Backend Technologies
```

markdownlint-cli2 (0.17.2)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)